src/workflow/CommandExecutor.py

```
@@ -254,7 +257,13 @@ def run_python(self, script_file: str, input_output: dict = {}) -> None:
             # run command without params
             self.run_command(["python", str(path)])
         elif isinstance(defaults, list):
-            defaults = {entry["key"]: entry["value"] for entry in defaults}
+            subsection = ''
+            defaults = {}
+            for entry in defaults:
+                if isinstance(entry, str):
+                    subsection = entry
+                    continue
+                defaults[f'{subsection}:{entry["key"]}'] = entry["value"]
```

Pylint (3.3.7)

[error] Iterated dict 'defaults' is being modified inside for loop body, iterate through a copy of it instead.

(E4702)